python/semantic_kernel/connectors/ai/content_filter_ai_exception.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, Dict, Optional
+
+from semantic_kernel.connectors.ai.ai_exception import AIException
+
+
+class ContentFilterAIException(AIException):
+    """AI exception for an error from Azure OpenAI's content filter"""
+
+    class ContentFilterCodes(Enum):
+        ResponsibleAIPolicyViolation = "ResponsibleAIPolicyViolation"
+
+    @dataclass
+    class ContentFilterResult:
+        class Severity(Enum):
+            High = "high"
+            Medium = "medium"
+            Safe = "safe"
+
+        filtered: bool = False
+        detected: bool = False
+        severity: Severity = Severity.Safe
+
+        @classmethod
+        def from_inner_error_result(
+            cls, inner_error_results: Dict[str, Any]
+        ) -> "ContentFilterAIException.ContentFilterResult":
+            """Creates a ContentFilterResult from the inner error results.
+
+            Arguments:
+                key {str} -- The key to get the inner error result from.
+                inner_error_results {Dict[str, Any]} -- The inner error results.
+
+            Returns:
+                ContentFilterResult -- The ContentFilterResult.
+            """
+            return cls(
+                filtered=inner_error_results.get("filtered", False),
+                detected=inner_error_results.get("detected", False),
+                severity=cls.Severity(
+                    inner_error_results.get("severity", cls.Severity.Safe.value)
+                ),
+            )
+
+    # The parameter that caused the error.
+    _param: str
+
+    # The error code specific to the content filter.
+    _content_filter_code: ContentFilterCodes
+
+    # The results of the different content filter checks.
+    _content_filter_result: Dict[str, ContentFilterResult]
+
+    def __init__(
```

Comment:
could we keep the signature the same between AIException and ContentFilterAIException, maybe add `**kwargs` to AIException and then use that for this as well, the init of this is mostly called by internal code, so kwargs are fine I think.

Reply:
Good call. Moved all the logic to the exception and left the signature the same as AIException.